client.go

```diff
@@ -110,7 +112,25 @@ func (d *Dialer) Dial(ctx context.Context, urlStr string, reqHdr http.Header) (*
 	}
 	req = req.WithContext(ctx)
 
-	rsp, err := d.RoundTripper.RoundTripOpt(req, http3.RoundTripOpt{DontCloseRequestStream: true})
+	rsp, err := d.RoundTripper.RoundTripOpt(req, http3.RoundTripOpt{
+		DontCloseRequestStream: true,
+		CheckSettings: func(settings http3.Settings) error {
+			if !settings.EnableExtendedConnect {
+				return errors.New("server didn't enable Extended CONNECT")
```

Comment:
Note that this check will break compatibility with older and the current version of webtransport-go, since quic-go servers only start enabling Extended CONNECT starting with the next release.

cc @sukunrt @MarcoPolo